scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/Job.java

```diff
@@ -324,17 +327,51 @@ public String display() {
     }
 
     /**
+     * Deadline can be as absolute or relative (from the job started time)
+     * Absolute deadline can be set as GI vairable `JOB_DDL` in ISO8601 date format
+     * without milliseconds, e.g. 2018-08-14T08:40:30+02:00.
+     * Relative deadline can be set as GI vairable `JOB_DDL` in the following format
+     * +HH:MM:SS, e.g. `+02:30:00`, or as +MM:SS, e.g. `+15:30`, or as +SS, e.g. `+40`.
      * @return job deadline if it exists
-     * (currently deadline can be set as job GI vairable in ISO8601 date format)
      */
-    public Optional<Date> getJobDeadline() {
+    public Optional<JobDeadline> getJobDeadline() {
         if (genericInformation.containsKey(JOB_DDL)) {
+            final String strJobDeadline = genericInformation.get(Job.JOB_DDL);
             try {
-                return Optional.of(ISO8601DateUtil.toDate(genericInformation.get(JOB_DDL)));
+                if (strJobDeadline.startsWith("+")) { // should be relative deadline
+                    return Optional.of(new JobDeadline(JobDeadline.parseDuration(strJobDeadline.substring(1)).get()));
+                } else { // should be absolute deadline
+                    return Optional.of(new JobDeadline(ISODateTimeFormat.dateTimeNoMillis()
+                                                                        .parseDateTime(strJobDeadline)
+                                                                        .toDate()));
+                }
             } catch (Exception e) {
-                LOGGER.warn("Imposible to parse JOB_DDL GI variable as ISO8601 date", e);
+                LOGGER.warn("Imposible to parse JOB_DDL GI variable as ISO8601 date: " + strJobDeadline);
             }
         }
+
         return Optional.empty();
     }
+
+    public static void main(String[] args) {
+        final Date date = ISODateTimeFormat.dateTimeNoMillis().parseDateTime("2230-10-20T19:45:00+01").toDate();
+
+        final Date date1 = ISODateTimeFormat.dateTimeNoMillis().parseDateTime("2230-10-20T19:45:00+02").toDate();
+
+        System.out.println(date.compareTo(date1));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)